src/main/java/uk/nhs/digital/nhsconnect/lab/results/inbound/NhsAckProducerService.java

```diff
@@ -0,0 +1,159 @@
+
+import com.github.mustachejava.Mustache;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.lab.results.mesh.message.WorkflowId;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Interchange;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.NhsAckContent;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.NhsAckStatus;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.NhsAckMessageContent;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.InterchangeHeader;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.MessageHeader;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.InterchangeParsingException;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.MessageParsingException;
+import uk.nhs.digital.nhsconnect.lab.results.sequence.SequenceService;
+import uk.nhs.digital.nhsconnect.lab.results.utils.TimestampService;
+import uk.nhs.digital.nhsconnect.lab.results.inbound.MessageProcessingResult.Success;
+import uk.nhs.digital.nhsconnect.lab.results.inbound.MessageProcessingResult.Error;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+import static uk.nhs.digital.nhsconnect.lab.results.utils.TemplateUtils.loadTemplate;
+import static uk.nhs.digital.nhsconnect.lab.results.utils.TemplateUtils.fillTemplate;
+
+@Component
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
+public class NhsAckProducerService {
+
+    @Autowired
+    private final TimestampService timestampService;
+
+    @Autowired
+    private final SequenceService sequenceService;
+
+    private static final int COMMON_SEGMENT_COUNT = 5;
+
+    private Mustache nhsAckTemplate = loadTemplate("nhsAck.mustache");
+
+    public String createNhsAckEdifact(WorkflowId workflowId, Interchange interchange,
+                                      List<MessageProcessingResult> messageProcessingResults) {
+
+        InterchangeHeader interchangeHeader = interchange.getInterchangeHeader();
+
+        var nhsAckContent = NhsAckContent.builder()
+            .interchangeSender(interchangeHeader.getSender())
+            .interchangeRecipient(interchangeHeader.getRecipient())
+            .interchangeControlReference(String.valueOf(interchangeHeader.getSequenceNumber()))
+            .nhsAckControlReference(String.valueOf(sequenceService.generateInterchangeSequence(
+                interchangeHeader.getRecipient(), interchangeHeader.getSender())))
+            .workflowId(workflowId.getWorkflowId())
+            .messageStatus(new ArrayList<>()).build();
+
+        int acceptedMessages = 0;
+        int rejectedMessages = 0;
+        int totalMessages = messageProcessingResults.size();
+
+        final String MESSAGE_ACCEPTED_STATUS_CODE = "MA";
```

Comment:
Can these be moved to class level and made static? Similar to `COMMON_SEGMENT_COUNT`